packages/core/src/operator/listenForChallenges.ts

```diff
@@ -11,41 +9,76 @@ import { resilientEventListener } from "../utils/resilientEventListener.js";
  * @param callback - The callback function to be triggered when ChallengeSubmitted event is emitted.
  * @returns A function that can be called to stop listening for the event.
  */
-export function listenForChallenges(callback: (challengeNumber: bigint, challenge: Challenge, event: any) => void): () => void {
-
-    // create a map to keep track of challengeNumbers that have called the callback
+export function listenForChallenges(callback: (challengeNumber: bigint, challenge: Challenge, event: any) => void, onError: (error: Error) => void): () => void {
```

Comment:
The `onError` shoud not be a required argument, not every place using this will need to act on websocket errors.

Reply:
Changed param to optional!

Reply:
Im not sure I follow, having the onError callback should not have impact on the error crashing or not. Having the try catch for the await is essential and that should print a log and we should create a ticket for better handling that within the operatorRuntime, but that should not be related to this ticket.

Let me know if that makes sense.

Reply:
I understand, this makes sense, thank you for the clarification.